### ISSUE TITLE
Handling multi-line items in DropdownButton and DropdownMenuItem

### ISSUE THREAD

**Issue by hundeva** [flutter/flutter#9211]
Internal: b/140830819

## Issue

The following [screenshot](http://i.imgur.com/78MzDCB.png) demonstrates the issue.

If the dropdown is open, the items are correctly displayed as - for example - two line items, but if the dropdown is closed, only the first line will be visible, and the dropdown indicator won't be visible.

## Steps to Reproduce

See the following test project:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(new MaterialApp(
    home: new Scaffold(
      appBar: new AppBar(
        title: new Text('Test project'),
      ),

      body: new Padding(
        padding: new EdgeInsets.all(16.0),
        child: new TestStatefulWidget(),
      ),
    ),
  ));
}

class TestStatefulWidget extends StatefulWidget {

  @override
  State<StatefulWidget> createState() => new TestStatefulWidgetState();
}

class TestStatefulWidgetState extends State<TestStatefulWidget> {

  static const List<String> shortItems = const [
    'These',
    'items',
    'are',
    'short',
    'enough',
  ];

  static const List<String> longItems = const [
    'These items will probably be longer than what fits on the screen correctly',
    'Because of this, the icon indicator on the right side will not show',
    'Instead, in development mode, a red indicator will display, showing something is wrong',
  ];

  String shortSpinnerValue = shortItems[0];
  String longSpinnerValue = longItems[0];

  @override
  Widget build(BuildContext context) {
    return new Column(
      mainAxisSize: MainAxisSize.min,
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        new DropdownButton(
          value: shortSpinnerValue,
          onChanged: (string) => setState(() => shortSpinnerValue = string),
          items: shortItems.map((string) {
            return new DropdownMenuItem(
              child: new Text(string),
              value: string,
            );
          }).toList(),
        ),

        new DropdownButton(
          value: longSpinnerValue,
          onChanged: (string) => setState(() => longSpinnerValue = string),
          items: longItems.map((string) {
            return new DropdownMenuItem(
              child: new Text(string),
              value: string,
            );
          }).toList(),
        ),
      ],
    );
  }
}
```

**Comment by abarth** [comment]
Would you expect it to wrap or to have an ellipsis?

**Comment by hundeva** [comment]
I'd expect an ellipsis, as in some corner cases, the selectable options may be very long (three, or even more lines is possible).

**Comment by eseidelGoogle** [comment]
If you just pass `overflow: TextOverflow.elipsis` to you Text call in you DropDownMenuItem I think you'll get the effect you want?

**Comment by hundeva** [comment]
Unfortunately, that won't work. If I add `overflow: TextOverflow.elipsis` to the text fields, the following thing happens: 
- [collapsed state](http://i.imgur.com/gO1ZA4q.png), nothing changed
- [expanded state](http://i.imgur.com/npMbgT8.png), the list items have been limited to be single line, which is not the desired effect, they should be completely visible, for example all 2 or 3 lines should be visible

The modified test project:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(new MaterialApp(
    home: new Scaffold(
      appBar: new AppBar(
        title: new Text('Test project'),
      ),

      body: new Padding(
        padding: new EdgeInsets.all(16.0),
        child: new TestStatefulWidget(),
      ),
    ),
  ));
}

class TestStatefulWidget extends StatefulWidget {

  @override
  State<StatefulWidget> createState() => new TestStatefulWidgetState();
}

class TestStatefulWidgetState extends State<TestStatefulWidget> {

  static const List<String> shortItems = const [
    'These',
    'items',
    'are',
    'short',
    'enough',
  ];

  static const List<String> longItems = const [
    'These items will probably be longer than what fits on the screen correctly',
    'Because of this, the icon indicator on the right side will not show',
    'Instead, in development mode, a red indicator will display, showing something is wrong',
  ];

  String shortSpinnerValue = shortItems[0];
  String longSpinnerValue = longItems[0];

  @override
  Widget build(BuildContext context) {
    return new Column(
      mainAxisSize: MainAxisSize.min,
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        new DropdownButton(
          value: shortSpinnerValue,
          onChanged: (string) => setState(() => shortSpinnerValue = string),
          items: shortItems.map((string) {
            return new DropdownMenuItem(
              child: new Text(
                string,
                overflow: TextOverflow.ellipsis,
              ),
              value: string,
            );
          }).toList(),
        ),

        new DropdownButton(
          value: longSpinnerValue,
          onChanged: (string) => setState(() => longSpinnerValue = string),
          items: longItems.map((string) {
            return new DropdownMenuItem(
              child: new Text(
                string,
                overflow: TextOverflow.ellipsis,
              ),
              value: string,
            );
          }).toList(),
        ),
      ],
    );
  }
}
```

**Comment by diegocatalao** [comment]
Hy @hundeva. I do not know if it will work for you but I think it's a good solution:

```dart
return new DropdownMenuItem<String>(
	value: item,
	child: new Container(
		width: 170.0,
		child: new Text(
			string,
			overflow: TextOverflow.ellipsis,
			style: new TextStyle(fontSize: 13.0)
		)
	)
);
```

Sorry to answer five months later and English too :sweat_smile:.
I hope it helps someone hehe.
Cheers.

**Comment by hundeva** [comment]
Hi,

Your english is fine, don't worry :)

The problem with that solution is, it doesn't really adapt to different screen sizes, where I would like it to fit the whole width for example. Of course, I could work around that like, the width of the Container is = width of the screen - width of the dropdown button, but that is a lot of boilerplate, which I would like to avoid.

Thanks!

**Comment by maryx** [comment]
I have the same question, except that I would like to show the full line, with word wrap, in the expanded state (since the user may not necessarily know what the line says if it ends in ellipses), but show ellipses in the collapsed state.
Currently, with this setup:
```
var units = <DropdownMenuItem>[];
    for (var unit in widget.units) {
      units.add(new DropdownMenuItem(
        value: unit.name,
        child: new Container(
          width: 130.0,
          child: new Text(
            unit.name,
            softWrap: true,
          ),
        ),
      ));
    }
```
I get this overlapping of text, where the wrapping of the words doesn't change the height of each dropdown menu item:
<img width="155" alt="screen shot 2017-10-22 at 8 44 32 pm" src="https://user-images.githubusercontent.com/3357543/31871562-15998aac-b76a-11e7-8dea-17ac87c527e2.png">

With a selected item, I get this:
<img width="133" alt="screen shot 2017-10-22 at 8 47 05 pm" src="https://user-images.githubusercontent.com/3357543/31871574-3072283e-b76a-11e7-9e48-b91546c84769.png">

**Comment by brianegan** [comment]
I think this may be a legit problem with the `DropDownButton` itself, and might have a simple fix... Happy to submit a PR if you think this is a good solution.

After looking at the `build` method of the `_DropdownButtonState`, it has an `IndexedStack` and `Icon` contained in a `Row`. However, since the `IndexedStack` is not wrapped in an `Expanded`, it is allowed to grow indefinitely. Therefore, even if you use `TextOverflow.ellipse` on your Text Widget, it will continue to Overflow.

I extracted the `DropDownButton` into a `MyDropdownButton` that wraps the `IndexedStack` with an `Expanded`. The Screenshots and code are below:

*Long Text Without Fix (No Expanded)*

<img width="487" alt="screen shot 2017-10-31 at 12 32 26 pm" src="https://user-images.githubusercontent.com/126604/32221920-a77e2e54-be37-11e7-95ab-526b1f6df6f7.png">

*Long Text With Fix*

<img width="487" alt="screen shot 2017-10-31 at 12 32 48 pm" src="https://user-images.githubusercontent.com/126604/32221938-b4ec67ae-be37-11e7-986a-b83b85aeb24b.png">

*Code with Fix*

```dart
    Widget result = new DefaultTextStyle(
      style: _textStyle,
      child: new SizedBox(
        height: widget.isDense ? _denseButtonHeight : null,
        child: new Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          mainAxisSize: MainAxisSize.min,
          children: <Widget>[
            // If value is null (then _selectedIndex is null) then we display
            // the hint or nothing at all.
            new Expanded(
              child: new IndexedStack(
                index: _selectedIndex ?? hintIndex,
                alignment: Alignment.centerLeft,
                children: items,
              ),
            ),
            new Icon(Icons.arrow_drop_down,
                size: widget.iconSize,
                // These colors are not defined in the Material Design spec.
                color: Theme.of(context).brightness == Brightness.light ? Colors.grey.shade700 : Colors.white70
            ),
          ],
        ),
      ),
    )
```

**Comment by maryx** [comment]
@brianegan You should submit a PR, it would help my current app a lot :). I'm not sure if it's the best solution but the reviewers will give feedback.
What does a dropdown item look like in that state? Is the text also truncated. or is the item height expanded?

**Comment by brianegan** [comment]
Hi @maryx :) The reason I didn't submit a PR just yet: If you use `TextOverflow.ellipsis`, the text will always be truncated, whether the dropdown is expanded or not, and the Widget works just fine.

If you need the text to run multiple lines, (by removing the `TextOverflow.ellipsis` and setting your text free), then you run into trouble, because the items have a fixed, hard-coded height.

Therefore, my solution felt helpful, but half-baked for such a fundamental Widget :)

**Comment by Hixie** [comment]
We should probably ask the material team what the right behaviour is here. Ellipsising the text doesn't seem ideal, since then the user can't read it (what if two items start the same way but differ at the end?). Wrapping seems against what the widget's core design is. WAP phones used to use a marquee effect to solve this...

**Comment by Hixie** [comment]
We don't have an ETA on this at the moment.

**Comment by maryx** [comment]
In the [Material Guidelines](https://material.io/guidelines/components/menus.html#menus-simple-menus) (search for "simple dialog instead"), it indicates that the lines should be wrapped. Could we replicate what the Simple Dialog does for the Dropdown widget?

**Comment by HansMuller** [comment]
The [Material Guidelines for menus](https://material.io/guidelines/components/menus.html#menus-simple-menus) says:

> If text in a simple menu wraps to a second line, use a simple dialog instead. Simple dialogs can have rows with varying heights.

I'll check with the Material team, but I would think that the same advice would apply to dropdowns.

**Comment by Hixie** [comment]
@HansMuller any news?

**Comment by Hixie** [comment]
We concluded that the quote @maryx posted above is what governs this for dropdowns as well.

We are not currently working on this, though it is on our radar. @maryx Can you speak to why this is a blocker, so we can prioritize it appropriately?

**Comment by maryx** [comment]
One of the Udacity apps is a Unit Converter. We have two dropdowns where user can select a unit to convert from, and a unit to convert to. Sometimes the name of the unit is longer than the width we've allocated for the dropdown. However:
1) The UX is changing, so we might end up with longer-width dropdowns, and this won't be a problem anymore
2) I'll try replacing the Dropdown items with SimpleDialog items, and that might solve the problem as well

I'll remove the `customer:udacity` tag for now. Thanks @Hixie for the update!

**Comment by Hixie** [comment]
@maryx Thanks for the update. Please don't hesitate to let us know if this becomes a blocker again, it would not be that difficult to fix if we need to (we just have lots of other things to fix too!).

**Comment by alexwhiteside1** [comment]
Hello! We're now running into this issue as well for one of our preferences screens, which allow users to choose options in context. Is there a work around for wrapping text in both collapsed and expanded states? I've tried adding expandeds, but still no luck.

**Comment by alexwhiteside1** [comment]
![screen shot 2018-04-17 at 15 54 32](https://user-images.githubusercontent.com/5420059/38902993-c0612062-4257-11e8-9a7d-20ffef0a550a.png)

**Comment by alexwhiteside1** [comment]
```dart  Widget transferPartnersView = new DropdownButtonHideUnderline(
        child: new DropdownButton<int>(
      iconSize: 0.0,
      value: widget.preferences["transferPartners"],
      onChanged: (int result) {
        setState(() {
          widget.preferences["transferPartners"] = result;
        });
      },
      items: <DropdownMenuItem<int>>[
        new DropdownMenuItem<int>(
          value: 0,
          child: new Text(transferPartners[0], style: BirchText.mediumLabel),
        ),
        new DropdownMenuItem<int>(
          value: 1,
          child: new Text(transferPartners[1], style: BirchText.mediumLabel),
        ),
      ],
    ));```

**Comment by pinkfish** [comment]
Running into this too.  Is there a work around to get the text clipped or something in the collapsed view?

**Comment by dyardyGIT** [comment]
Running into this too.  I would prefer to allow wrapping.

**Comment by bluelinemx** [comment]
Running into this too. I prefer allow wrapping to.

**Comment by roNn23** [comment]
Also running into this issue... :/

**Comment by Shantanu-Kotambkar** [comment]
We ran into this issue too. Any update on this team.

**Comment by Shantanu-Kotambkar** [comment]
We ran into this issue. Any update on this team?

**Comment by Shantanu-Kotambkar** [comment]
@brianegan The work around you suggested above, I am not really able to understand it. Are you replacing the Dropdown button entirely with your custom code?

**Comment by brianegan** [comment]
@Shantanu-Kotambkar Yep!

**Comment by Shantanu-Kotambkar** [comment]
@brianegan Your answer on StackOverflow proved to be helpful, posting the link here so that others may review if needed [https://stackoverflow.com/questions/47032262/flutter-dropdownbutton-overflow-in-listview](url). 

However, I am facing another issue now, I do not want my Drop down menu option to be truncated, as the user won't be able to read it. Right now, both my selected Text and my Drop Down menu items are getting truncated. Is there any way to be able to display the DropDown menu its completely and just the selected option truncated. Am I doing something wrong here? Was the fix supposed to do the behavior I am describing?

**Comment by brianegan** [comment]
@Shantanu-Kotambkar Not with my solution, unfortunately. That's definitely why I consider it a half-baked solution that might not work for everyone (https://github.com/flutter/flutter/issues/9211#issuecomment-340936947).

**Comment by Shantanu-Kotambkar** [comment]
Thanks @brianegan .  Is there any update on this issue @maryx?

**Comment by Sdlion** [comment]
Faced the same problem
I wouldn't mind the text get cropped on the option selected.
But in the expanded view I need to show my customer the whole options, but the height of each `DropDownMenuItem` seems fixed.
With a `Container` as the child of the menu item I can control the drop down menu's width and of each menu item. But setting the `height` attribute won't change `DropDownMenuItem` behavior as the `width` or `constrains` attribute.

**Comment by KingDoxik** [comment]
I am currently facing this issue as well. Hope, there will be a fix soon.

**Comment by apgapg** [comment]
The solution provided by @brianegan on stackoverflow breaks when you wrap FixDropDownButton in a Row. I have fixed this problem in my CustomDropDown, use this file:
https://github.com/apgapg/Flutter_Pull_Request/blob/master/fix_drop_down.dart

**Comment by Shantanu-Kotambkar** [comment]
@apgapg Does your implementation support multiple modes:
 1. Display full text in the dropdown menu without cropping. 
 2. Crop the text to fit the size of the textfield in which the menu option is displayed when selected.

**Comment by apgapg** [comment]
@Shantanu-Kotambkar  Yes. The full text is displayed in the dropdown menu, ie the dropdown item becomes more than 1 line by default. Also when you select this , because the dropdown selection has to be single line, it automatically crops the multilinetext to single line. I will attach screenshot for reference

**Comment by Shantanu-Kotambkar** [comment]
That will be awesome @apgapg

**Comment by Shantanu-Kotambkar** [comment]
@apgapg It doesn't work for me the way you said. For me the text displayed in dropdown menu item is still cropped. Can you provide a sample implementation and screenshot?

**Comment by apgapg** [comment]
The menu item will get cropped if the number of lines are greater than 2, because there is a height constraint of 48.0 on menu items.

**Comment by apgapg** [comment]
@Shantanu-Kotambkar I created a sample : https://github.com/apgapg/flutter_drop_down.
Will attach screenshot soon

**Comment by Shantanu-Kotambkar** [comment]
@apgapg I get it, you are altering hint-text value instead of setting up the value of dropdown item. It works for my use case, but it is a hack, and we still need to work on resolving the issue. Great work though! Thanks for the work around.

**Comment by MarcoLude** [comment]
We ran into this issue too. Any update on this?

**Comment by mklim** [comment]
For now the new [`isExpanded` flag](https://github.com/flutter/flutter/blob/9a290c174a8d039460239779a21175271245824d/packages/flutter/lib/src/material/dropdown.dart#L538) may double as a workaround or partial fix for this. With it set to true longer lines are wrapped in both closed and opened states.

Not sure if that's a real fix for this underlying issue though. It does still break if the overflow grows past two lines, and it's probably not ideal for line wrapping to be coupled to the arrow icon placement. I can imagine cases where somebody would want potential long text to wrap but also want the dropdown to be collapsed whenever possible.

I'm going to track down the answers to a few questions before I try to put in a more robust fix for this. Earlier in this issue it's mentioned that long text should be wrapped, but I'm thinking there's probably still some kind of upper limit for what `DropdownButton` should expand to fit. Assuming that's true, I'll need to know what the cutoff point should be, and how it should handle truncating the extra text. And if the button should also wrap the text in the unselected state.

**Comment by Sdlion** [comment]
> I'm going to track down the answers to a few questions before I try to put in a more robust fix for this. Earlier in this issue it's mentioned that long text should be wrapped, but I'm thinking there's probably still some kind of upper limit for what `DropdownButton` should expand to fit. Assuming that's true, I'll need to know what the cutoff point should be, and how it should handle truncating the extra text. And if the button should also wrap the text in the unselected state.

@mklim In my opinion, it shouldn't exceed more than 2-3 lines. Or more broadly: 

1. There should be on-screen visual cues that it's a multiple choice menu (more than one visible and clear separation).
2. If the same options are presented as radio text options, the distance between dividing rulers (item separation lines) shouldn't be too far from the vertical aligned radio. Style consistency.

This naturally implies that there should be a height/line max limit; thus long text should gracefully fall to be truncated. The trimming point has been represented by convention through ellipses ('HORIZONTAL ELLIPSIS' (U+2026)).

To show longer text one could consider a different approach. Like a whole new View where the different options could be shown. This could be noted in the documentation to explain the Widget limitations and alternatives. Thing is, if the widget or the configuration parameters fail, they should fail gracefully.

**Comment by diego-lipinski-de-castro** [comment]
2 years, still persists

**Comment by rajyadavnp** [comment]
The easiest solution is to add `isExpanded: true` in `DropdownButton`:

For example:

```
DropdownButtonHideUnderline(
  child: ButtonTheme(
    child: DropdownButton(
      isExpanded: true, //Add This line and done.
      items: <DropdownMenuItem>[
        DropdownMenuItem(
          child: Text('All accounts'),
          value: 0,
        ),
        DropdownMenuItem(child: Text('My accounts'), value: 1),
        DropdownMenuItem(
          child: Text('My business 1'),
          value: 2,
        )
      ],
      value: 0,
    ),
  ),
)
```

**Comment by razvangeangu** [comment]
`isExpanded: true` is not implemented for the `DropdownButtonFormField` though

**Comment by olaide-ekolere** [comment]
I just cannot help but wonder why this dropdownbutton doesnt work well, decided to avoid it like death and use my own custom code for this. Thats my advice to everyone.

**Comment by YAPPDEF** [comment]
Also running into this issue.
If I add isExpanded: true, the dropdown doesn't have a render issue, but in the non expanded case (selected value), not both lines are shown (height with cut off of the second line).

**Comment by andersonmendesdev** [comment]
Any solution? I have this problem with DropdownButtonFormField.

**Comment by miguelpruivo** [comment]
Facing this in a `DropdownButtonFormField` as well. The flag `isExpanded` is not currently exposed to it.

**Comment by shihaohong** [comment]
There's currently a PR for exposing `isExpanded` to `DropdownButtonFormField`: https://github.com/flutter/flutter/pull/36998, so that issue should be resolved once that PR lands.

In general, I think there is some pain with figuring out how to handle multiple lines for `DropdownButton` and `DropdownMenuItem`. I'll try to see if I can get some guidance on how this should be handled according to the Material guidelines, since [the dropdown menu spec](https://material.io/design/components/menus.html#specs) currently provides values assuming single-line items.

![spec](https://user-images.githubusercontent.com/27032613/63372163-6ca84f80-c33a-11e9-9800-d1c2a69169c0.png)

**Comment by andywolff** [comment]
+1 for finding a solution which would let dropdown buttons and items grow to fit wrapped text. My team's app has dropdowns which look fine in English with default text size, but wrap to 2-3 lines in largest text mode (a11y) or more verbose languages (i18n), which causes them to get cut off.

**Comment by Hixie** [comment]
Given that the material design specification doesn't say how to render multiline dropdown widgets, it's not clear how we should address this.

**Comment by jkurtw** [comment]
b/140830819

**Comment by eseidelGoogle** [comment]
@willlarche tends to find answers when MD doesn't have directions. :)

**Comment by jkurtw** [comment]
may cut off important text in large font modes or locales more verbose than English, but because the issue is not present for the button, users can see any clipped text by selecting it and seeing it in the button

**Comment by willlarche** [comment]
We're working on making the guidance clearer in the future about handling text scaling. Thanks for your patience. And the diagrams in that article contain bugs that make this even harder to understand.

Here's what should be built:

1. Default vertical padding is .5 * font.pointSize. 
2. Content is centered and can be multiline with default spacing between lines.
3. We suggest that you cap at 2 lines and truncate after that but there's no reason that has to be logic that's baked into the component. It can be flexible.

At most flexible, this should be a list that a developer can pass anything into and handle the content and padding and everything themselves. At easiest to use, the should be able to pass in a Text and get baseline Material styling.

Am I forgetting anything?

**Comment by malviyaritesh** [comment]
@git-xbits If you know to use [`DropdownButtonFormField`](https://api.flutter.dev/flutter/material/DropdownButtonFormField/DropdownButtonFormField.html), then `MultilineDropdownButtonFormField` is just a replacement of it with added properties like `isExpanded` and `isDense`. 

To use it, you just need to replace `DropdownButtonFormField` with `MultilineDropdownButtonFormField` and provide properties like `isExpanded: true` & `isDense: false`. 

Here's a basic example (incomplete to work) showing its uses:

```dart
// ...
// Other widgets
// ...
MultilineDropdownButtonFormField<int>(
  isExpanded: true,
  isDense: false,
  items: [
    DropdownMenuItem(
      child: Text(
        'Very looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong first option',
        overflow: TextOverflow.ellipsis,
        maxLines: 2,
        softWrap: true,
      ),
      value: 1,
    ),
    DropdownMenuItem(
      child: Text(
        'Very looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong second option',
        overflow: TextOverflow.ellipsis,
        maxLines: 2,
        softWrap: true,
      ),
      value: 2,
    ),
    DropdownMenuItem(
      child: Text(
        'Very looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong third option',
        overflow: TextOverflow.ellipsis,
        maxLines: 2,
        softWrap: true,
      ),
      value: 3,
    )
  ],
  decoration: InputDecoration(
    labelText: 'Transportation',
    icon: Icon(Icons.assignment),
  ),
  value: _dropdownValue,
  onChanged: (int value) {
    setState(() {
      _selectedDropdownValue = value;
    });
  },
  onSaved: (int value) => _dropdownValue = value,
),
// ...
// Other widgets
// ...
```

**Comment by HansMuller** [comment]
This [version of the original example](https://gist.github.com/HansMuller/85eaf23771ca71c271e5b6446838ad14) demonstrates how one can use the [new selectedItemBuilder](https://github.com/flutter/flutter/pull/40461) feature to customize the way the dropdown's selected item is displayed - and avoid the overflow. It also shows how one can hew to the Material Design spec by limiting dropdown items to two lines of text. Note also that the dropdown button's `isExpanded` parameter is true in the second dropdown, to force the dropdown's button to fit within the available horizontal space.

| Closed | Open |
| --- | --- |
| ![menu_closed](https://user-images.githubusercontent.com/1377460/65173772-658d6380-da04-11e9-9c78-778747646e67.png) | ![menu_open](https://user-images.githubusercontent.com/1377460/65173779-68885400-da04-11e9-8d6f-cb7a0d7dade7.png) |

**Comment by HansMuller** [comment]
I'm going to close this issue because the original problem has been addressed: https://github.com/flutter/flutter/issues/9211#issuecomment-532803519

Dropdown menus are still limited because their items are fixed height. That came up in some of the feedback, for example https://github.com/flutter/flutter/issues/9211#issuecomment-526331605, https://github.com/flutter/flutter/issues/9211#issuecomment-526331605, https://github.com/flutter/flutter/issues/9211#issuecomment-413087329.

I've opened a separate issue for the fixed height limitation: https://github.com/flutter/flutter/issues/40773

**Comment by miguelpruivo** [comment]
@HansMuller this doesn't seem to be available for `DropdownButtonFormField`. Am I missing something?

**Comment by HansMuller** [comment]
@miguelpruivo - you are correct, https://github.com/flutter/flutter/issues/43021

**Comment by shihaohong** [comment]
The `DropdownButtonFormField.selectedItemBuilder`'s equivalent has also been created as well: https://github.com/flutter/flutter/issues/42481

**Comment by darkstarx** [comment]
Вот понаписали тонны слов. Лучше бы просто исправили. Стыд программистам Google!!!

**Comment by GSPrakashNaidu** [comment]
> I have the same question, except that I would like to show the full line, with word wrap, in the expanded state (since the user may not necessarily know what the line says if it ends in ellipses), but show ellipses in the collapsed state.
> Currently, with this setup:
> 
> ```
> var units = <DropdownMenuItem>[];
>     for (var unit in widget.units) {
>       units.add(new DropdownMenuItem(
>         value: unit.name,
>         child: new Container(
>           width: 130.0,
>           child: new Text(
>             unit.name,
>             softWrap: true,
>           ),
>         ),
>       ));
>     }
> ```
> 
> I get this overlapping of text, where the wrapping of the words doesn't change the height of each dropdown menu item. Ideally, each DropdownMenuItem should space according to its text wrap:
> <img alt="screen shot 2017-10-22 at 8 44 32 pm" width="155" src="https://user-images.githubusercontent.com/3357543/31871562-15998aac-b76a-11e7-8dea-17ac87c527e2.png">
> 
> With a selected item, I get this. Ideally, I would be able to set the `overflow` to ellipses only for the currently-selected dropdown item in collapsed view. DropdownButton only lets me set a TextStyle though:
> <img alt="screen shot 2017-10-22 at 8 47 05 pm" width="133" src="https://user-images.githubusercontent.com/3357543/31871574-3072283e-b76a-11e7-9e48-b91546c84769.png">


Thank you @maryx :), It worked for me.
